modules/update-upgrading-mcp.adoc

```diff
@@ -0,0 +1,234 @@
+// Module included in the following assemblies:
+//
+// * storage/persistent_storage/persistent-storage-hostpath.adoc
+
+[id="update-upgrading-mcp_{context}"]
+= Delaying the update for specific nodes
+
+There might be situations where you do not want to update specific nodes when you update a cluster. Updating nodes requires the nodes to be cordoned and drained, which marks the nodes as unschedulable and evicts all pods. For example, you can ensure that critical applications stay available during the update process.
+
+In these cases, you can use machine config pools (MCPs) to efffectively delay updating specific nodes until you are ready to cordon and drain those nodes. Delaying the updates in this manner is referred to as *in-service software updates (ISSU)* or *canary updates*.
+
+If you have specific nodes that you do not want to cordon and drain at the time of an update, before the update, create one or more machine config pools (MCPs) and move the nodes from their current MCP to one of the new MCPs. Then, pause the new MCPs. The nodes associated with the paused MCPs do not get cordoned, drained, or updated. When you are ready to update those nodes, unpause the new MCPs, which starts the update process on those nodes. After the nodes are updated, verify that your applications are functioning properly. Then, move the nodes back to their original MCP, so that the nodes receive any configuration changes made to those MCPs. If you have more than one custom MCP, you can update one set of nodes at a time, to ensure everything is functioning properly before updating the other nodes.
```

Comment:
Per [original description](https://github.com/openshift/openshift-docs/pull/34445/files#diff-72d7c105477b4165dbe6af361e4e379589a17c7302264aa164669f6528aab1faR32) and [workflow](https://github.com/openshift/openshift-docs/pull/34445/files#diff-72d7c105477b4165dbe6af361e4e379589a17c7302264aa164669f6528aab1faR71), the application validation should be happened on the first unpaused and updated mcp nodes. Here and the following procedure sounds like it is guide user validate application after all nodes are updated. I do not think that is a right work flow for canary updates.

Reply:
> In these cases, before the update, create one or more machine config pools (MCPs) and move the nodes from their current MCP to one of the new MCPs. Then, pause the new MCPs. The nodes associated with the paused MCPs do not get cordoned, drained, or updated. After the cluster is updated, when you are ready to update those nodes, unpause one of the new MCPs, which starts the update process on those nodes. After those nodes are updated, verify that your applications are functioning properly. Then, move the nodes back to their original MCP, so that the nodes receive any configuration changes made to those MCPs. Then, update the remaining MCPs one-by-one until all of the nodes are updated.

Reply:
The follow step 4 and step 5 need to be switched over to keep consistent with this description,  am I right?

Reply:
What if we add a step? 
4. Unpause one MCP.
5. Test the applications.
**_6. If successful, unpause any other MCPs and test the applications._**
7. Move the nodes to the original MCP.

Reply:
The following example, only introduce one customized mcp, the ideal case is something like:
1. user's application running on all the nodes belonging to 'worker' mcp
2. create a user customized mcp ('node-noupdate'), and pause this mcp
3. trigger upgrade
4. Test the applications running on the nodes belonging to 'worker' mcp that was already was upgraded.
5. if successful, unpause the user customized mcp, and test the application running on the nodes belonging to 'node-noupdate' mcp.
6. Move the nodes to the original MCP.